### PR TITLE
chore: Clarify wrapping-key aws-kms behavior for non-discovery mode

### PR DESCRIPTION
**Pull request by lavaleri**
*Issue #, if available:*

*Description of changes:*
We've had customer confusion around this point. We are working to clarify the AWS Docs, but we should also clarify in readthedocs here.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by lavaleri** [comment]
Note from June that saying something like "configured ARN" could be confusing for customers, as they could then think that a "Configured ARN" is a thing they need to understand. Instead, consider rewording to make it explicitly "the ARN you configured the MKP with".

File: README.rst [hunk 1]
```diff
@@ -205,6 +205,8 @@ These parameters are common to all master key providers:
   key provider. ``The discovery`` attribute is only available if you are using an ``aws-kms`` provider.
 
     * If using ``aws-kms`` to decrypt, `you must specify either a key or discovery with a value of true`_.
+    * If using ``aws-kms`` to decrypt and specifying a key, decryption will only be attempted for keys identified with
+      a key ARN.
```
Comment (farleyb-amazon) [inline README.rst:209]:
```suggestion
    * If using ``aws-kms`` to decrypt and specifying a key, you must use a key ARN; key ids, alias names, and alias ARNs are not supported.
```
I found this a bit hard to parse. I would use the same wording you use in the other code bases.

**Review by farleyb-amazon — APPROVED** [review]
